### DISCUSSION TITLE
What is the meaning of the "row_value" in the HighsSolution when solving a MIP model

### DISCUSSION THREAD

**Discussion by linpeng0105** [ERGO-Code/HiGHS#1975] [Q&A] [answered]
This is a c++ example from the call_highs_from_cpp.cpp
```
  for (int row = 0; row < lp.num_row_; row++)
  {
    cout << "Row    " << row << " " << lp.row_names_[row];
    if (info.primal_solution_status)
      cout << "; value = " << solution.row_value[row];
    cout << endl;
  }
```

When solving a MIP model, what is the meaning of the "row_value" in the HighsSolution

**Comment by jajhall** [comment, marked as answer]
The vector of row values is $Ax$, where $A$ is the constraint matrix, and $x$ is the vector of column (ie solution) values.

**Reply by linpeng0105** [reply]
Thank you very much!

**Comment by linpeng0105** [comment]
One more question, please~

When solving MIP, I want to set a dual-bound for a model before and during the solving, which interfaces should I call?
